src/components/course/tests/CourseRunCard.test.jsx

```diff
@@ -174,6 +195,29 @@ describe('<CourseRunCard/>', () => {
     expect(screen.getByText(enrollButtonTypes.HIDE_BUTTON)).toBeTruthy();
   });
 
+  test('User must request enrollment', () => {
+    // With Browse/Request feature flag on and mock-enabled for the customer,
+    // the user should only see a Request Enrollment button if they
+    // have no license or offers.
+    config.features.FEATURE_BROWSE_AND_REQUEST = true;
+    const courseRun = generateCourseRun({});
+    const noUserSubsidyState = {
+      subscriptionLicense: null,
+      offers: {
+        offers: [],
+        offersCount: 0,
+      },
+    };
+    renderCard({
+      courseRun,
+      initialUserSubsidyState: noUserSubsidyState,
+    });
+    const startDate = moment(COURSE_RUN_START).format(DATE_FORMAT);
+    expect(screen.getByText(`Starts ${startDate}`)).toBeTruthy();
+    expect(screen.getByText('Be the first to enroll!')).toBeTruthy();
+    expect(screen.getByText(enrollButtonTypes.HIDE_BUTTON)).toBeTruthy();
```

Comment:
nit: the `expect(...).toBeTruthy()` is unnecessary when using `screen.getBy*`. If the text isn't found, `screen.getBy*` will already fail the test.

```jsx
screen.getByText(`Starts ${startDate}`)
// ...
```